pxr/usdValidation/usdShadeValidators/testenv/testUsdShadeValidators.cpp

```diff
@@ -554,13 +558,223 @@ TestUsdShadeEncapsulationRulesValidator()
+    const UsdValidationValidator *validator =
+        registry.GetOrLoadValidatorByName(
+        UsdShadeValidatorNameTokens->normalMapTextureValidator);
+    TF_AXIOM(validator);
+
+    // Create a Stage, Material, and Two Shaders (UsdPreviewSurface,
+    // UsdUVTexture).
+    UsdStageRefPtr usdStage = UsdStage::CreateInMemory();
+    UsdShadeMaterial material = UsdShadeMaterial::Define(usdStage,
+        SdfPath("/RootMaterial"));
+
+    const std::string usdPreviewSurfaceShaderPath =
+        "/RootMaterial/UsdPreviewSurface";
+    UsdShadeShader usdPreviewSurfaceShader = UsdShadeShader::Define(
+        usdStage, SdfPath(usdPreviewSurfaceShaderPath));
+    usdPreviewSurfaceShader.CreateIdAttr(
+        VtValue(TfToken("UsdPreviewSurface")));
+    UsdPrim usdPreviewSurfaceShaderPrim = usdPreviewSurfaceShader.GetPrim();
+
+    UsdShadeShader usdUvTextureShader = UsdShadeShader::Define(
+        usdStage, SdfPath("/RootMaterial/NormalTexture"));
+    usdUvTextureShader.CreateIdAttr(VtValue(TfToken("UsdUVTexture")));
+
+    // Add initial valid file and sourceColorSpace input values.
+    std::string textureAssetPath = "./normalMap.jpg";
+    UsdShadeInput fileInput = usdUvTextureShader.CreateInput(
+        TfToken("file"), SdfValueTypeNames->Asset);
+    fileInput.Set(SdfAssetPath(textureAssetPath));
+    UsdShadeInput sourceColorSpaceInput = usdUvTextureShader.CreateInput(
+        TfToken("sourceColorSpace"), SdfValueTypeNames->Token);
+    const TfToken rawToken("raw");
+    sourceColorSpaceInput.Set(rawToken);
+
+    // Connect the output of the UsdUVTexture Shader to the normal of the
+    // UsdPreviewSurface Shader.
+    usdUvTextureShader.CreateOutput(TfToken("rgb"), SdfValueTypeNames->Float3);
+    UsdShadeInput normalInput = usdPreviewSurfaceShader.CreateInput(
+        TfToken("normal"), SdfValueTypeNames->Normal3f);
+    normalInput.ConnectToSource(
+        SdfPath("/RootMaterial/NormalTexture.outputs:rgb"));
+
+    // Verify invalid bias & scale error, they should exists and do
+    // not exist at this point.
+    UsdValidationErrorVector errors = validator->Validate(
+        usdPreviewSurfaceShaderPrim);
+    TfToken expectedErrorIdentifier(
+    "usdShadeValidators:NormalMapTextureValidator.NonCompliantBiasAndScale");
+    std::string expectedErrorMsg =
+        TfStringPrintf("UsdUVTexture prim <%s> reads 8 bit Normal Map "
+                           "@./normalMap.jpg@, which requires that "
+                           "inputs:scale be set to (2, 2, 2, 1) and "
+                           "inputs:bias be set to (-1, -1, -1, 0) for proper "
+                           "interpretation as per the UsdPreviewSurface and "
+                           "UsdUVTexture docs.",
+                           usdUvTextureShader.GetPath().GetText());
+    ValidateError(errors,
+        expectedErrorIdentifier,
+        usdUvTextureShader.GetPath(),
+        expectedErrorMsg);
```

Comment:
You can scope this such that the expected* variables are const, as is done in other testing methods here.
```cpp
{
    const UsdValidationErrorVector errors = validator->...;
    const TfToken expectedErrorIdentifier("...");
    const std::string expectedErrorMsg = TfStringPrintf("...");
    ValidateErrors(errors, expectedErrorIdentifier, usdUvTextureShader.GetPath(), expectedErrorMsg);
}
```

Reply:
Same for below.